src/doc/src/reference/resolver.md

```diff
@@ -91,6 +93,31 @@ at the time of this writing) and Package B will use the greatest `0.6` release
 (`0.6.5` for example). This can lead to potential problems, see the
 [Version-incompatibility hazards] section for more details.
 
+The resolver algorithm favors the greatest available versions of dependencies
+and has limited support for backtracking in the solutions it attempts.
+Therefore, it may converge on a solution that includes two copies of a
+dependency when one would suffice. For example:
```

Comment:
I suggest moving this example down as a separate sub-section of the Troubleshooting section. I don't think it is something that is something users *should* encounter regularly, and demonstrates a requirement style that is not recommended (and is explicitly discouraged). This is pretty early in the chapter to start discussing some relatively rare aspects.

Reply:
Done.